app/uk/gov/hmrc/gform/models/FormModelBuilder.scala

```diff
@@ -104,6 +97,9 @@ class FormModelBuilder[E, F[_]: Functor](
       case ExpressionResult.StringResult(value) => VariadicFormData.one[SourceOrigin.Current](modelComponentId, value)
       case ExpressionResult.OptionResult(value) =>
         VariadicFormData.many[SourceOrigin.Current](modelComponentId, value.map(_.toString))
+      case ExpressionResult.DateResult(_) =>
+        VariadicFormData
+          .empty[SourceOrigin.Current] //TODO: geneate DateResult instead of StringResult for ExprType DateString
```

Comment:
This implementation looks suspicious. Why `empty`? Is it relate to previous comment?

Reply:
correct. How to represent LocalDate as string as VariadicValue only supports String right?

Reply:
oh, yes, we don't want to put `DateResult` into `VariadicFormData`, so `empty` is correct

Reply:
I have modified to put a string representation of Date into VariadicFormData (yyyymmdd). That should be ok I guess?